py_gtfs_rt_ingestion/batch_files.py

```diff
@@ -95,9 +71,42 @@ def make_file_list(args: BatchArgs) -> Iterable[(str, int)]:
         total_gigs = total_bytes / 1000000000
         logging.info("Batched %d gigs across %d files" % (total_gigs,
                                                           total_files))
+    else:
+        print(json.dumps([b.create_event() for b in batches], indent=2))
+
+
+def lambda_handler(event: dict, context) -> None:
+    """
+    AWS Lambda Python handled function as described in AWS Developer Guide:
+    https://docs.aws.amazon.com/lambda/latest/dg/python-handler.html
+    :param event: The event dict sent by Amazon API Gateway that contains all of
+            the request data.
+    :param context: The context in which the function is called.
+    :return: A response that is sent to Amazon API Gateway, to be wrapped into
+             an HTTP response. The 'statusCode' field is the HTTP status code
+             and the 'body' field is the body of the response.
+
+    expected event structure is
+    {
+        s3_prefix: str
+        s3_bucket: str
+        filesize_threshold: int
+        pretty: bool
+    }
+
+    batch files should all be of same ConfigType
+    """
+    logging.info("Processing event:\n%s" % json.dumps(event, indent=2))
+    try:
+        batch_args = BatchArgs(**event)
+        logging.info(batch_args)
+        main(batch_args)
+    except Exception as e:
+        # log if something goes wrong and let lambda recatch the exception
+        logging.exception(e)
+        raise e
```

Comment:
keeping this function slim. convert the args from a dict to a BatchArgs object (which will throw if the event isn't 1:1 with the members of a BatchArgs instance) and run main.

i wanted to wrap this in a try / except so that our logs will get a full traceback of any error. i re-raise for now to let lambda handle the exception and communicate back to us. we might want to change this.